src/components/GenerateKeypair.tsx

```diff
@@ -0,0 +1,58 @@
+import { Button, Icon, Input } from "@stellar/design-system";
+import { Keypair } from "stellar-sdk";
+import { useStore } from "@/store/useStore";
+
+export const GenerateKeypair = () => {
+  const { account } = useStore();
+
+  const generateKeypair = () => {
+    let keypair = Keypair.random();
+
+    account.updateKeypair({
+      publicKey: keypair.publicKey(),
+      secretKey: keypair.secret(),
+    });
+  };
+
+  return (
+    <div className="Account__keypair">
+      <Button size="md" variant="tertiary" onClick={generateKeypair}>
+        Generate keypair
+      </Button>
+
+      {account.keypair.publicKey && (
+        <Input
+          readOnly
+          id="generate-keypair-publickey"
+          fieldSize="md"
+          label="Public Key"
+          value={account.keypair.publicKey}
+          rightElement={
+            <Icon.Copy01
+              onClick={() => {
+                navigator.clipboard.writeText(account.keypair.publicKey);
+              }}
+            />
+          }
```

Comment:
`Input` component has a built-in copy button (including copy functionality). :sunglasses: You can see an example of it in `NetworkSelector` component. So we can remove any extra dependencies that were installed.

```suggestion
          copyButton={{
            position: "right",
          }}
```